generic/tests/mac_change.py

```diff
@@ -1,31 +1,38 @@
 import re
 import logging
 
-from autotest.client.shared import error
-
+from virttest import error_context
 from virttest import utils_misc
 from virttest import utils_net
 from virttest import utils_test
 
 
-def check_guest_mac(mac, vm, device_id=None):
-    error.context("Check mac address via monitor", logging.info)
-    network_info = vm.monitor.info("network")
+@error_context.context_aware
+def check_guest_mac(test, mac, vm, device_id=None):
+    """
+    check mac address of guest via qmp.
+
+    :param test: test object
+    :param mac: mac address of guest
+    :param vm: target vm
+    :param device_id: id of network pci device
+    """
+    error_context.context("Check mac address via monitor", logging.info)
+    network_info = str(vm.monitor.info("network"))
     if not device_id:
         device_id = vm.virtnet[0].device_id
 
-    if device_id not in str(network_info):
-        err = "Could not find device '%s' from query-network monitor command."
-        err += "query-network command output: %s" % str(network_info)
-        raise error.TestFail(err)
-    for info in str(network_info).splitlines():
-        if re.match(device_id, info.strip(), re.I) and mac not in info:
-            err = "Cold not get correct mac from qmp command!"
-            err += "query-network command output: %s" % str(network_info)
-            raise error.TestFail(err)
+    if device_id not in network_info:
+        err = "Could not find device '%s' from query-network monitor command.\n"
+        err += "query-network command output: %s" % network_info
+        test.error(err)
+    if device_id in network_info.strip() and mac not in network_info:
```

Comment:
Here, it wants to check if the `mac` is owned by the given nic (`device_id`), so I think the more accurate approach should be:
```python
if not re.search(("%s.*%s" % (device_id, mac)), network_info, re.M | re.I):
    err = ...
```

Reply:
Yes, you're right. I will update it, thanks!